contrib/bsp/src/mill/contrib/bsp/MillBuildServer.scala

```diff
@@ -371,8 +375,10 @@ class MillBuildServer(evaluator: Evaluator, bspVersion: String, serverVersion: S
         .foldLeft(Seq.empty[ScalacOptionsItem]) { (items, targetId) =>
           val newItem =
             if (targetId == millBuildTargetId) {
-              val classpath = Try(getClass.getClassLoader.asInstanceOf[SpecialClassLoader])
-                .fold(_ => Seq.empty, _.allJars.filter(url => exists(Path(url.getFile))).map(_.toURI.toString))
+              val classpath = Try(getClass.getClassLoader.asInstanceOf[SpecialClassLoader]).fold(
+                _ => Seq.empty,
+                _.allJars.filter(url => !isSourceJar(url) && exists(Path(url.getFile))).map(_.toURI.toString)
```

Comment:
We are filtering out the source jars here since they are loaded in `buildTargetDependencySources`.